src/sentry/utils/auth.py

```diff
@@ -243,7 +244,9 @@ def find_users(username, with_valid_password=True, is_active=None):
     return []
 
 
-def login(request, user, passed_2fa=None, after_2fa=None, organization_id=None, source=None):
+def login(
+    request: HttpRequest, user, passed_2fa=None, after_2fa=None, organization_id=None, source=None
```

Comment:
Should we type out the rest of these attrs too?

Reply:
I typed request since i wanted to distinguish request as a HttpRequest over the alternative Request class. I wouldn't mind typing out the rest of those attrs tho